lib/heap_sort.rb

```diff
@@ -1,8 +1,22 @@
-
+require_relative "min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  O(n log n)
+# Space Complexity: O(n log n)
+def heapsort(list)
```

Comment:
👍 , however the space complexity is O(n) because you're building a heap of n elements.